### PR TITLE
Remove bad conventions from Feature Variants Doc

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
Removes bad practices in Example 1 and 6 on https://docs.gradle.org/current/userguide/feature_variants.html, showing feature variants with the _main_ source set.

### CODE REVIEW

**Comment by lkasso** [comment]
@bot-gradle test BD

**Comment by lkasso** [comment]
@jvandort Do you want me to make additional changes to the page?

**Comment by jvandort** [comment]
@bot-gradle test bd

**Comment by jvandort** [comment]
I have made some changes and pushed a new commit. PTAL and let me know what you think.

I have also rebased this PR onto master since there were some merge conflicts

**Comment by jvandort** [comment]
@bot-gradle test this

**Review by jvandort — APPROVED** [review]
LGTM other than the merge failure

File: platforms/documentation/docs/src/docs/userguide/dep-man/04-modeling-features/feature_variants.adoc [hunk 1]
```diff
@@ -100,12 +101,21 @@ Furthermore, consumable configurations are created in the same manner as for the
 - the configuration `mongodbSupportApiElements`, used by consumers to fetch the artifacts and API dependencies of this feature
 - the configuration `mongodbSupportRuntimeElements`, used by consumers to fetch the artifacts and runtime dependencies of this feature
 
+<<<<<<< HEAD:subprojects/docs/src/docs/userguide/dep-man/04-modeling-features/feature_variants.adoc
```
[the remark this answers is not in the document]
Comment (jvandort) [reply platforms/documentation/docs/src/docs/userguide/dep-man/04-modeling-features/feature_variants.adoc:104]: See step 5, 6 here: 
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

**Comment by lkasso** [comment]
@bot-gradle test and merge

**Pull request merged**